pkg/controller/backup/backup_controller.go

```diff
@@ -0,0 +1,119 @@
+package backup
+
+import (
+	"time"
+
+	hivemetrics "github.com/openshift/hive/pkg/controller/metrics"
+	controllerutils "github.com/openshift/hive/pkg/controller/utils"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	controllerName = "backup"
+)
+
+// Add creates a new Backup Controller and adds it to the Manager with default RBAC. The Manager will set fields on the
+// Controller and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return AddToManager(mgr, NewReconciler(mgr))
+}
+
+// NewReconciler returns a new reconcile.Reconciler
+func NewReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileBackup{
+		Client: controllerutils.NewClientWithMetricsOrDie(mgr, controllerName),
+		scheme: mgr.GetScheme(),
+		logger: log.WithField("controller", controllerName),
+	}
+}
+
+// AddToManager adds a new Controller to mgr with r as the reconcile.Reconciler
+func AddToManager(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New(controllerName+"-controller", mgr, controller.Options{Reconciler: r, MaxConcurrentReconciles: controllerutils.GetConcurrentReconciles()})
+	if err != nil {
+		return err
+	}
+
+	reconciler := r.(*ReconcileBackup)
+
+	clusterDeploymentBackup := newClusterDeploymentBackup(reconciler)
+	err = clusterDeploymentBackup.registerClusterDeploymentWatch(c)
+	if err != nil {
+		return err
+	}
+
+	return nil // If we got here, then no errors occurred.
+}
+
+// This ensures that ReconcileBackup struct implements all functions that the reconcile.Reconciler interface requires.
+var _ reconcile.Reconciler = &ReconcileBackup{}
+
+// ReconcileBackup ensures that Velero backup objects are created when changes are made to Hive objects.
+type ReconcileBackup struct {
+	client.Client
+	scheme *runtime.Scheme
+
+	logger log.FieldLogger
+}
+
+// Reconcile ensures that all Hive object changes have corresponding Velero backup objects.
+// ClusterDeployment.Spec.Config.Machines
+// +kubebuilder:rbac:groups=hive.openshift.io,resources=clusterdeployments,verbs=get;list;watch
+// +kubebuilder:rbac:groups=velero.io,resources=backups,verbs=create
+func (r *ReconcileBackup) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	start := time.Now()
+	contextLogger := addReconcileRequestLoggerFields(r.logger, request)
```

Comment:
IMO we can just call the `contextLogger` as `logger` as it will be more easy for the eyes to read. If someone is interested to know how logger is working they can find how it is implemented.

Reply:
+1

Reply:
I believe I got the pattern of calling the newer logger `contextLogger` from @dgoodwin. 

I believe the idea is to distinguish it from the standard logger.

I'm fine changing this, but want @dgoodwin to weigh in first.

Reply:
I'm big on contextual loggers but can't recall using it as a var name, though it's entirely possible. Typically we try to come up with a short name that also reflects what the context the logger is carrying, ex cdLog, or in this case perhaps nsLog. "logger" would be fine as well but slight pref to indicating what's in there.